OpenEdXMobile/src/main/java/org/edx/mobile/util/VideoUtil.java

```diff
@@ -47,9 +49,36 @@ public static boolean isSupportedVideoFormat(@NonNull String videoUrl, @NonNull
      * @return <code>true</code> if video is downloadable, <code>false</code> otherwise.
      */
     public static boolean isVideoDownloadable(@NonNull VideoData video) {
+        return getPreferredVideoUrlForDownloading(video) != null;
+    }
+
+    /**
+     * Gives the preferred video url for downloading from a specified video data.
+     *
+     * @param video Data of the video whose preferred download url is needed.
+     * @return download url of video, return null if unable to find any suitable url for downloading.
+     */
+    @Nullable
+    public static String getPreferredVideoUrlForDownloading(@NonNull VideoData video) {
         final VideoInfo preferredVideoInfo = video.encodedVideos.getPreferredVideoInfo();
-        return preferredVideoInfo != null &&
-                !isSupportedVideoFormat(preferredVideoInfo.url, VIDEO_FORMAT_M3U8) &&
-                !video.onlyOnWeb;
+        if (preferredVideoInfo == null || video.onlyOnWeb) {
+            return null;
+        }
+        if (!isVideoHasFormat(preferredVideoInfo.url, VIDEO_FORMAT_M3U8)) {
+            return preferredVideoInfo.url;
+        }
+        if (!new Config(MainApplication.instance()).isUsingVideoPipeline()) {
+            /**
```

Comment:
This isn't a javadoc rather a devnote. So we should use `/*` in this case instead of `/**`

Reply:
Redirections for e.g. {@link DownloadEntry#url} only work in /** , java doc.